src/main/java/com/redislabs/redisgraph/impl/GraphCacheList.java

```diff
@@ -0,0 +1,67 @@
+package com.redislabs.redisgraph.impl;
+
+import com.redislabs.redisgraph.Record;
+import com.redislabs.redisgraph.RedisGraph;
+import com.redislabs.redisgraph.ResultSet;
+
+import java.util.ArrayList;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Represents a local cache of list of strings. Holds data from a specific procedure, for a specific graph.
+ */
+public class GraphCacheList extends ArrayList<String> {
```

Comment:
I wouldn't extend the list but make it an internal field

Reply:
done